site/public/js/ta-grading-rubric.js

```diff
@@ -1867,11 +1886,27 @@ function onComponentOrderChange() {
  * @param me DOM element of the cancel button
  */
 function onCancelOverallComment(me) {
-    toggleOverallComment(false)
-        .catch(function (err) {
+  // if the overall Comment is changed then prompt the TA for loss in comment
+  ajaxGetOverallComment(getGradeableId(), getAnonId()).then((lastSavedOverallComment) => {
```

Comment:
This function is almost identical to the above, lets have a helper function somewhere that accepts the getGradedComponent promise and live custom comment and compares them, if they're different give a prompt otherwise do nothing.

That way both these methods can call that function and in the future we need to change/add functionality it'll be easier.

Reply:
@shailpatels the two methods just look identical but they use different DOM components, so I guess writing a helper function for just checking the equality in comments (which is currently done inside if condition) will be too much. What do you think?

Reply:
Oh ok I see now, I guess its not ideal but if theres not a good way to consolidate code then this is good, I'll approve now.

Reply:
Thanks!